app/controllers/sites_controller.rb

```diff
@@ -1,8 +1,17 @@
 class SitesController < ApplicationController
-  before_action :authenticate
+  before_action :authenticate, :build_provider
 
   def index
-    @provider = Provider.includes(:sites).find(params[:provider_code]).first
     @sites = @provider.sites.sort_by(&:location_name)
   end
+
+  def edit
+    @site = Site.where(provider_code: params[:provider_code]).find(params[:id]).first
```

Comment:
`Provider.includes(:sites)` should get you all the sites, so I don't think you need another API call here, right?

Reply:
Wouldn't it be better to only get the site necessary for the edit action, instead of all the sites? In which case `build_provider` would be limited to only trigger for the `index` action.

Reply:
We need the provider to render the page as well, which is why I suggested the first option.